Assorted build fixes

dlespiau opened the pull request:
I somewhat doubt we've had make distcheck running recently, but this series is a leap towards it.

The (last?) thing remaining is being able to run all docker tests under a non-root user (already captured in an issue).

There's an awkward thing in having the docker tests part of make check (and so make distcheck): it means we need to run have docker running and configured to use the compiled runtime (problably through make install and having docker configured to take the runtime from the installation prefix)

This series also tweaks the travis build to do out-of-tree builds out of an exploded make dist tarball to add extra tests (for instance that we do distribute all the necessary files in the tarball). That's close to what make distcheck does without actually doing a make distcheck (that can't run in travis).

dlespiau commented:
Fixes https://github.com/01org/cc-oci-runtime/issues/323

dlespiau commented:
Fixes https://github.com/01org/cc-oci-runtime/issues/79

dlespiau commented:
PR updated:

  - Rebased against latest master (was conflicting)
  - Fixed https://github.com/01org/cc-oci-runtime/issues/277
  - make distcheck now passes on my laptop

dlespiau commented:
Fixes https://github.com/01org/cc-oci-runtime/issues/277
Fixes https://github.com/01org/cc-oci-runtime/issues/322

jodh-intel commented on .travis/travis-setup-test-directory.sh:
```diff
@@ -0,0 +1,28 @@
+#
+# Note: we don't use `make distcheck` here as we can't run everything we want
+#       for discheck in travis.
+chronic make dist
```
Might be worth a comment saying why `chronic(1)` is being used here.

jodh-intel commented on Makefile.am:
```diff
@@ -127,7 +130,8 @@ DISTCHECK_CONFIGURE_FLAGS = \
 	--enable-code-coverage \
-	--enable-functional-tests
+	--enable-functional-tests \
+	--with-systemdsystemunitdir=/tmp/cc-distcheck/systemdunitdir/
```
Can this directory be made unique?

dlespiau replied:
Actually, I didn't really want to make it unique: it's not needed (ie. not needing several distinct copies of that data) and doing several make distcheck will not end up clobbering /tmp.

jodh-intel replied:
Maybe a comment to that effect would be worthwhile to explain this.

dlespiau replied:
That I can do!

dlespiau commented:
Updated PR:
  - Rebased against master (was conflicting with recent developments)
  - Put the build steps inside their own shell script to work around limitations in travis
  - Added some comments about the use chronic
  - Added quoting to $tarball_dir. Note that It's not actually a problem and adding quotes may actually fail to alert us that the dist tarball has a path with a space in, which isn't a nice property for a dist tarball. I'm not going to bikeshed the bikeshed though and just added the quotes.
  - Added a comment about the systemd unit dir in make distcheck

grahamwhaley commented:
Nice series @dlespiau  I think maybe we are waiting for just one or two tiny addends from @jodh-intel feedback, but other than that,
lgtm

dlespiau commented:
Fixed the "builddi" typo in commit message.

All the feedback from James should have been addressed, hopefully.

jodh-intel commented on .travis.yml:
```diff
@@ -22,13 +22,13 @@ env:
 
 before_script:
-- ./.travis-setup.sh
+- ./.travis/travis-setup.sh
```
The leading `./` is redundant, but innocuous :-)

jodh-intel commented:
lgtm

grahamwhaley commented:
I don't think the leading ./ will block the merge, so:
lgtm

chavafg commented:
qa-failed

dlespiau commented:
Looking at the CI dashboard, it shows that make distcheck passes (yey!) but make check doesn't (from time to time). It seems like it's a known flaky functional test:

```
not ok 42 start and kill state
```

jodh-intel commented:
surely it's :snowflake: ? :smile:

chavafg commented:
qa-passed

dlespiau commented:
@jodh-intel I wish!, ok merging that one. Thanks all for the reviews.
